### PR TITLE
Feat/bitfinex exchange user stream

### PR DESCRIPTION
**Pull request by malinich**
**Before submitting this PR, please make sure**:
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/", etc)

**Optional**:
- Related Github issue:
- Related Clubhouse Story:

**A description of the changes proposed in the pull request**:
Second MR for Bitfinex exchange user-stream.
 Include:
- User-stream data source
- User-stream tracker
- Bitfinex auth

### CODE REVIEW

File: hummingbot/market/bitfinex/bitfinex_api_user_stream_data_source.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    async def listen_for_user_stream(self, ev_loop: asyncio.BaseEventLoop, output: asyncio.Queue):
+        while True:
+            try:
+                ws: websockets.WebSocketClientProtocol
+                async with websockets.connect(BITFINEX_WS_URI) as ws:
+                    ws: websockets.WebSocketClientProtocol = ws
+                    payload = self._bitfinex_auth.generate_auth_payload()
+                    await ws.send(json.dumps(payload))
+                    await asyncio.wait_for(ws.recv(), timeout=self.MESSAGE_TIMEOUT)  # info
+                    await asyncio.wait_for(ws.recv(), timeout=self.MESSAGE_TIMEOUT)  # auth
```
Comment (CrimsonJacket) [inline hummingbot/market/bitfinex/bitfinex_api_user_stream_data_source.py]: Considering that we will be processing the message after `_get_response()` and that the response from the initial subscription can be determined, I feel that this is not necessary.

File: hummingbot/market/bitfinex/bitfinex_api_user_stream_data_source.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                self.logger().error(
+                    "Unexpected error with Bitfinex WebSocket connection. " "Retrying after 30 seconds...",
+                    exc_info=True,
+                )
+                await asyncio.sleep(self.MESSAGE_TIMEOUT)
+
+    def _transform_message_from_exchange(self, raw_msg) -> BitfinexOrderBookMessage:
+        msg = ujson.loads(raw_msg)
+        order_book_message: BitfinexOrderBookMessage = BitfinexOrderBook.diff_message_from_exchange(msg, time.time())
+        return order_book_message
```
Comment (CrimsonJacket) [inline hummingbot/market/bitfinex/bitfinex_api_user_stream_data_source.py]: We should handle the type of messages here

File: hummingbot/market/bitfinex/bitfinex_order_book_message.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    def trade_id(self) -> int:
+        return self.content["trade_id"]
+
+    @property
+    def trading_pair(self) -> str:
+        return self.content["symbol"]
+
+    @property
+    def type_hb(self):
+        return self.content[1] == "hb" if isinstance(self.content, list) else None
```
Comment (CrimsonJacket) [inline hummingbot/market/bitfinex/bitfinex_order_book_message.py]: Small issue, but renaming this `type_heartbeat` would help understand the function

File: hummingbot/market/bitfinex/bitfinex_api_user_stream_data_source.py (hunk 2 above)
Comment (malinich) [reply hummingbot/market/bitfinex/bitfinex_api_user_stream_data_source.py]:
```
@property
def type_heartbeat(self):
```

File: hummingbot/market/bitfinex/bitfinex_api_user_stream_data_source.py (hunk 1 above)
Comment (malinich) [reply hummingbot/market/bitfinex/bitfinex_api_user_stream_data_source.py]: removed these rows. Now process them in `_transform_message_from_exchange`

File: hummingbot/market/bitfinex/bitfinex_order_book_message.py (hunk 3 above)
Comment (malinich) [reply hummingbot/market/bitfinex/bitfinex_order_book_message.py]: rename in behalf of more humane
```
@property
def type_heartbeat(self):
```

**Comment by CrimsonJacket** [comment]
test this please

**Comment by CrimsonJacket** [comment]
test this please

**Pull request merged**